File: src/protocol/BlindsDOWNCommand.cpp
#include "BlindsDOWNCommand.h"
#include "CommandVisitor.h"

using namespace std;
using namespace communication;


BlindsDOWNCommand::BlindsDOWNCommand()
    : Command(10, CommandType::BLINDS_DOWN)
{}

BlindsDOWNCommand::~BlindsDOWNCommand()
{}

vector<uint8_t> BlindsDOWNCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

string BlindsDOWNCommand::getName()
{
    return string("BlindsDOWNCommand");
}

void BlindsDOWNCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void BlindsDOWNCommand::initializeDataSize()
{
    setDataSize(sizeof(commandType_));
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/BlindsDOWNCommand.java
package wpam.mobile_client.protocol;

import java.util.ArrayList;

public class BlindsDOWNCommand extends Command
	{
		public BlindsDOWNCommand()
		{
			super(10, CommandType.BLINDS_DOWN);
		}
		public final void close()
		{
			super.close();
		}

		@Override
		public ArrayList<Integer> getFrameBytes()
		{
			initializeDataSize();

			ArrayList<Integer> frame = getHeader();
			frame.add(commandType_.getValue() & 0xFF);

			return frame;
		}
		@Override
		public String getName()
		{
			return new String("BlindsDOWNCommand");
		}
		@Override
		public void accept(CommandVisitor visitor)
		{
			visitor.visit(this);
		}

		@Override
		protected void initializeDataSize()
		{
			setDataSize(1);
		}
	}